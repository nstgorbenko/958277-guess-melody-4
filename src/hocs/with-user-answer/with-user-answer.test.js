import withUserAnswer from "./with-user-answer.js";

import React from "react";
import renderer from "react-test-renderer";

const mockQuestion = {
  type: `genre`,
  genre: `rock`,
  answers: [{
    src: `https://upload.wikimedia.org/wikipedia/commons/4/4e/BWV_543-fugue.ogg`,
    genre: `rock`,
  }, {
    src: `https://upload.wikimedia.org/wikipedia/commons/4/4e/BWV_543-fugue.ogg`,
    genre: `blues`,
  }, {
    src: `https://upload.wikimedia.org/wikipedia/commons/4/4e/BWV_543-fugue.ogg`,
    genre: `jazz`,
  }, {
    src: `https://upload.wikimedia.org/wikipedia/commons/4/4e/BWV_543-fugue.ogg`,
    genre: `rock`,
  }],
};

const MockComponent = () => (<div></div>);

describe(`WithUserAnswer HOC rendering`, () => {
  it(`renders correctly`, () => {
    const MockComponentWrapped = withUserAnswer(MockComponent);

    const tree = renderer
      .create(
          <MockComponentWrapped question={mockQuestion}
            userAnswers={[false, true, false, false]}
            onAnswer={() => {}}
            onChange={() => {}}
          />)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
