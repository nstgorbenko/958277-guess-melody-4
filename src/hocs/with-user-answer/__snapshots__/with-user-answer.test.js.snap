// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WithUserAnswer HOC rendering renders correctly 1`] = `<div />`;
